[2023-10-02 17:58:16,993: INFO: main: Welcome to our custom logging]
[2023-10-02 18:50:48,715: INFO: 2805191277: >>>>>> stage Data Ingestion stage started <<<<<<]
[2023-10-02 18:50:48,755: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 18:50:48,767: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 18:50:48,775: ERROR: 2805191277: yaml file is empty]
Traceback (most recent call last):
  File "c:\users\preet\mlops_project\src\mlProject\utils\common.py", line 32, in read_yaml
    return ConfigBox(content)
  File "box\box.py", line 259, in box.box.Box.__init__
box.exceptions.BoxValueError: First argument must be mapping or iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\preet\AppData\Local\Temp\ipykernel_24016\2805191277.py", line 20, in <module>
    obj.main()
  File "C:\Users\preet\AppData\Local\Temp\ipykernel_24016\2805191277.py", line 8, in main
    config = ConfigurationManager()
  File "C:\Users\preet\AppData\Local\Temp\ipykernel_24016\2582655453.py", line 9, in __init__
    self.params = read_yaml(params_filepath)
  File "c:\Users\preet\mlops_project\venv\lib\site-packages\ensure\main.py", line 849, in __call__
    return_val = self.f(*args, **kwargs)
  File "c:\users\preet\mlops_project\src\mlProject\utils\common.py", line 34, in read_yaml
    raise ValueError("yaml file is empty")
ValueError: yaml file is empty
[2023-10-02 18:52:12,349: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 18:52:12,359: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 18:53:40,637: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 18:53:40,673: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 18:53:40,685: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 18:53:40,694: INFO: common: created directory at: artifacts]
[2023-10-02 18:53:40,701: INFO: common: created directory at: artifacts/data_ingestion]
[2023-10-02 18:53:41,928: INFO: 2280763589: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 6030:28FA:A67BC:C7ED1:651AC44E
Accept-Ranges: bytes
Date: Mon, 02 Oct 2023 13:23:41 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10232-MAA
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1696253021.075342,VS0,VE508
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: ab9d6c71913d966847b69f862a3cd84aa4c77ce3
Expires: Mon, 02 Oct 2023 13:28:41 GMT
Source-Age: 0

]
[2023-10-02 19:08:19,340: INFO: main: >>>>>> stage data ingestion started <<<<<<]
[2023-10-02 19:08:19,352: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 19:08:19,358: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 19:08:19,362: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 19:08:19,364: INFO: common: created directory at: artifacts]
[2023-10-02 19:08:19,368: INFO: common: created directory at: artifacts/data_ingestion]
[2023-10-02 19:08:19,374: INFO: data_ingestion: File already exists of size: ~ 23 KB]
[2023-10-02 19:08:19,398: INFO: main: >>>>>> stage data ingestion completed <<<<<<

x==========x]
[2023-10-02 19:08:40,084: INFO: main: >>>>>> stage data ingestion started <<<<<<]
[2023-10-02 19:08:40,096: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 19:08:40,101: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 19:08:40,106: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 19:08:40,113: INFO: common: created directory at: artifacts]
[2023-10-02 19:08:40,117: INFO: common: created directory at: artifacts/data_ingestion]
[2023-10-02 19:08:41,067: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 6030:28FA:A67BC:C7ED1:651AC44E
Accept-Ranges: bytes
Date: Mon, 02 Oct 2023 13:38:40 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10229-MAA
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1696253920.426015,VS0,VE282
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 9cac573a87af12d145e815eac453a41fc9b70068
Expires: Mon, 02 Oct 2023 13:43:40 GMT
Source-Age: 0

]
[2023-10-02 19:08:41,136: INFO: main: >>>>>> stage data ingestion completed <<<<<<

x==========x]
[2023-10-02 21:10:13,091: INFO: main: >>>>>> stage data ingestion started <<<<<<]
[2023-10-02 21:10:13,133: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:10:13,145: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:10:13,166: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:10:13,172: INFO: common: created directory at: artifacts]
[2023-10-02 21:10:13,176: INFO: common: created directory at: artifacts/data_ingestion]
[2023-10-02 21:10:14,239: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 6030:28FA:A67BC:C7ED1:651AC44E
Accept-Ranges: bytes
Date: Mon, 02 Oct 2023 15:40:13 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10220-MAA
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1696261213.480368,VS0,VE285
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: f0b1aa9fde13079525b516d393dda13025f04b8b
Expires: Mon, 02 Oct 2023 15:45:13 GMT
Source-Age: 0

]
[2023-10-02 21:10:14,306: INFO: main: >>>>>> stage data ingestion completed <<<<<<

x==========x]
[2023-10-02 21:17:03,022: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:17:03,031: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:17:03,050: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:17:03,059: INFO: common: created directory at: artifacts]
[2023-10-02 21:17:03,066: INFO: common: created directory at: artifacts/data_validation]
[2023-10-02 21:21:18,042: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2023-10-02 21:21:18,061: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:21:18,067: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:21:18,079: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:21:18,084: INFO: common: created directory at: artifacts]
[2023-10-02 21:21:18,089: INFO: common: created directory at: artifacts/data_validation]
[2023-10-02 21:21:18,155: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2023-10-02 21:34:58,252: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:34:58,262: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:34:58,321: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:34:58,329: INFO: common: created directory at: artifacts]
[2023-10-02 21:34:58,339: INFO: common: created directory at: artifacts/data_transformation]
[2023-10-02 21:35:24,385: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:35:24,393: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:35:24,409: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:35:24,417: INFO: common: created directory at: artifacts]
[2023-10-02 21:35:24,425: INFO: common: created directory at: artifacts/data_transformation]
[2023-10-02 21:35:24,559: INFO: 290957180: Splited data into training and test sets]
[2023-10-02 21:35:24,566: INFO: 290957180: (1199, 12)]
[2023-10-02 21:35:24,571: INFO: 290957180: (400, 12)]
[2023-10-02 21:50:35,316: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2023-10-02 21:50:35,332: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:50:35,340: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:50:35,361: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:50:35,368: INFO: common: created directory at: artifacts]
[2023-10-02 21:50:35,373: INFO: common: created directory at: artifacts/data_ingestion]
[2023-10-02 21:50:36,330: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 6030:28FA:A67BC:C7ED1:651AC44E
Accept-Ranges: bytes
Date: Mon, 02 Oct 2023 16:20:35 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10228-MAA
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1696263636.524026,VS0,VE291
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 77d79bb4f75b515f600b608c14a3a6f49b7ee7dc
Expires: Mon, 02 Oct 2023 16:25:35 GMT
Source-Age: 0

]
[2023-10-02 21:50:36,390: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2023-10-02 21:50:36,394: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2023-10-02 21:50:36,413: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:50:36,418: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:50:36,448: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:50:36,453: INFO: common: created directory at: artifacts]
[2023-10-02 21:50:36,457: INFO: common: created directory at: artifacts/data_validation]
[2023-10-02 21:50:36,460: ERROR: main: 'NoneType' object has no attribute 'unzip_data_dir']
Traceback (most recent call last):
  File "C:\Users\preet\mlops_project\main.py", line 26, in <module>
    data_ingestion.main()
  File "c:\users\preet\mlops_project\src\mlProject\pipeline\stage_02_data_validation.py", line 16, in main
    data_validation.validate_all_columns()
  File "c:\users\preet\mlops_project\src\mlProject\components\data_validation.py", line 35, in validate_all_columns
    raise e
  File "c:\users\preet\mlops_project\src\mlProject\components\data_validation.py", line 16, in validate_all_columns
    data = pd.read_csv(self.config.unzip_data_dir)
AttributeError: 'NoneType' object has no attribute 'unzip_data_dir'
[2023-10-02 21:52:53,587: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2023-10-02 21:52:53,622: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:52:53,628: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:52:53,643: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:52:53,649: INFO: common: created directory at: artifacts]
[2023-10-02 21:52:53,659: INFO: common: created directory at: artifacts/data_ingestion]
[2023-10-02 21:52:54,352: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 6030:28FA:A67BC:C7ED1:651AC44E
Accept-Ranges: bytes
Date: Mon, 02 Oct 2023 16:22:53 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10234-MAA
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1696263774.838272,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 1dd16eb2111173258293df1611b4bb0f00a05a94
Expires: Mon, 02 Oct 2023 16:27:53 GMT
Source-Age: 138

]
[2023-10-02 21:52:54,405: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2023-10-02 21:52:54,418: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2023-10-02 21:52:54,485: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:52:54,533: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:52:54,593: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:52:54,629: INFO: common: created directory at: artifacts]
[2023-10-02 21:52:54,634: INFO: common: created directory at: artifacts/data_validation]
[2023-10-02 21:52:54,636: ERROR: main: 'NoneType' object has no attribute 'unzip_data_dir']
Traceback (most recent call last):
  File "C:\Users\preet\mlops_project\main.py", line 26, in <module>
    data_ingestion.main()
  File "c:\users\preet\mlops_project\src\mlProject\pipeline\stage_02_data_validation.py", line 16, in main
    data_validation.validate_all_columns()
  File "c:\users\preet\mlops_project\src\mlProject\components\data_validation.py", line 35, in validate_all_columns
    raise e
  File "c:\users\preet\mlops_project\src\mlProject\components\data_validation.py", line 16, in validate_all_columns
    data = pd.read_csv(self.config.unzip_data_dir)
AttributeError: 'NoneType' object has no attribute 'unzip_data_dir'
[2023-10-02 21:54:17,709: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:54:17,719: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:54:17,738: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:54:17,747: INFO: common: created directory at: artifacts]
[2023-10-02 21:54:17,754: INFO: common: created directory at: artifacts/data_ingestion]
[2023-10-02 21:54:18,184: INFO: 2280763589: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 6030:28FA:A67BC:C7ED1:651AC44E
Accept-Ranges: bytes
Date: Mon, 02 Oct 2023 16:24:17 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10243-MAA
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1696263858.663983,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 30ef1681c121ef2baf18991e7d2ecd16e4e8d6c4
Expires: Mon, 02 Oct 2023 16:29:17 GMT
Source-Age: 222

]
[2023-10-02 21:56:50,507: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:56:50,516: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:56:50,534: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:56:50,542: INFO: common: created directory at: artifacts]
[2023-10-02 21:56:50,550: INFO: common: created directory at: artifacts/data_validation]
[2023-10-02 21:57:18,646: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 21:57:18,656: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 21:57:18,678: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 21:57:18,686: INFO: common: created directory at: artifacts]
[2023-10-02 21:57:18,696: INFO: common: created directory at: artifacts/data_transformation]
[2023-10-02 21:57:18,832: INFO: 290957180: Splited data into training and test sets]
[2023-10-02 21:57:18,838: INFO: 290957180: (1199, 12)]
[2023-10-02 21:57:18,842: INFO: 290957180: (400, 12)]
[2023-10-02 22:00:09,428: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2023-10-02 22:00:09,446: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 22:00:09,451: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 22:00:09,470: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 22:00:09,474: INFO: common: created directory at: artifacts]
[2023-10-02 22:00:09,477: INFO: common: created directory at: artifacts/data_validation]
[2023-10-02 22:00:09,547: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2023-10-02 22:00:34,385: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2023-10-02 22:00:34,403: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-02 22:00:34,408: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-02 22:00:34,422: INFO: common: yaml file: schema.yaml loaded successfully]
[2023-10-02 22:00:34,428: INFO: common: created directory at: artifacts]
[2023-10-02 22:00:34,431: INFO: common: created directory at: artifacts/data_transformation]
[2023-10-02 22:00:34,545: INFO: data_transformation: Splited data into training and test sets]
[2023-10-02 22:00:34,546: INFO: data_transformation: (1199, 12)]
[2023-10-02 22:00:34,548: INFO: data_transformation: (400, 12)]
[2023-10-02 22:00:34,550: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
